src/styles/blocks/stars.scss

```diff
@@ -0,0 +1,22 @@
+.stars {
+  display: flex;
+  align-items: center;
+
+  @for $i from 1 through 5 {
+    &--#{$i} &__star:nth-child(-n + #{$i}) {
```

Comment:
The selector `&--#{$i} &__star:nth-child(-n + #{$i})` is incorrect. The `&--#{$i}` should be a modifier of the `.stars` class, so it should be `.stars--#{$i} .stars__star:nth-child(-n + #{$i})` to correctly apply the styles to the child elements.